pkg/network/setup/netstat.go

```diff
@@ -90,60 +90,71 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		return nil
 	}
 
-	multusStatusNetworksByName := netvmispec.IndexInterfaceStatusByName(
+	oldIfaceStatusesByName := netvmispec.IndexInterfaceStatusByName(
 		vmi.Status.Interfaces,
 		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
-			return netvmispec.ContainsInfoSource(ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus)
+			return ifaceStatus.PodInterfaceName != "" ||
+				netvmispec.ContainsInfoSource(ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus)
 		},
 	)
 	vmiInterfacesSpecByName := netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
 
-	interfacesStatus := ifacesStatusFromDomainInterfaces(domain.Spec.Devices.Interfaces)
-	interfacesStatus = append(interfacesStatus,
+	updatedIfaceStatuses := ifacesStatusFromDomainInterfaces(domain.Spec.Devices.Interfaces)
+	updatedIfaceStatuses = append(updatedIfaceStatuses,
 		sriovIfacesStatusFromDomainHostDevices(domain.Spec.Devices.HostDevices, vmiInterfacesSpecByName)...,
 	)
 
 	var err error
-	interfacesStatus, err = c.updateIfacesStatusFromPodCache(interfacesStatus, vmi.Spec.Domain.Devices.Interfaces, vmi)
+	updatedIfaceStatuses, err = c.updateIfacesStatusFromPodCache(updatedIfaceStatuses, vmi.Spec.Domain.Devices.Interfaces, vmi)
 	if err != nil {
 		return err
 	}
 
 	// Guest Agent information will add and conditionally override data gathered from the cache.
-	interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, domain.Status.Interfaces)
+	updatedIfaceStatuses = ifacesStatusFromGuestAgent(updatedIfaceStatuses, domain.Status.Interfaces)
 
-	primaryInterfaceStatus, interfacesStatus := netvmispec.PopInterfaceByNetwork(interfacesStatus, netvmispec.LookupPodNetwork(vmi.Spec.Networks))
+	updatedIfaceStatuses = restoreVirtCtrlOwnedInfo(updatedIfaceStatuses, oldIfaceStatusesByName, vmiInterfacesSpecByName)
+
+	primaryInterfaceStatus, updatedIfaceStatuses := netvmispec.PopInterfaceByNetwork(updatedIfaceStatuses, netvmispec.LookupPodNetwork(vmi.Spec.Networks))
 	if primaryInterfaceStatus != nil {
-		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
+		updatedIfaceStatuses = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, updatedIfaceStatuses...)
 	}
 
-	interfacesStatus = ifacesStatusFromMultus(interfacesStatus, multusStatusNetworksByName, vmiInterfacesSpecByName)
-
-	vmi.Status.Interfaces = interfacesStatus
+	vmi.Status.Interfaces = updatedIfaceStatuses
 
 	c.removeAbsentIfacesFromVolatileCache(vmi)
 
 	return nil
 }
 
-func ifacesStatusFromMultus(
-	interfacesStatus []v1.VirtualMachineInstanceNetworkInterface,
-	multusStatusNetworksByName map[string]v1.VirtualMachineInstanceNetworkInterface,
+func restoreVirtCtrlOwnedInfo(
```

Comment:
I think this is worth explaining what and why it does it. It is not trivial for a reader and I am sure we our-self will not remember much in a year from now.

Reply:
This change was reverted.